TheThingsNetwork/lorawan-stack
# Remove support for NaN and Infinity in decoded payloads

#### Summary



Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/6128

#### Changes



- Decoded payloads which contain `NaN` or `Infinity` values cause the payload decoding to fail instead of emitting a warning.
- Remove decoded payload invalid field value metrics.

#### Testing



Local testing. This can be tested on `staging1` as follows:

1. Create an application and an end device. Contents don't matter - we simulate traffic.
2. Set the uplink formatter to:
```js
function decodeUplink(input) {
  return {
    data: {
      bytes: input.bytes,
      nan: NaN,
      inf: Infinity,
    },
    warnings: [],
    errors: []
  };
}
```
3. Simulate an uplink.
4. Expect that payload decoding fails with the following error:
```json
{
    "@type": "type.googleapis.com/ttn.lorawan.v3.ErrorDetails",
    "namespace": "pkg/messageprocessors/javascript",
    "name": "output_encoding",
    "message_format": "{errors}",
    "attributes": {
        "errors": ".nan: invalid NaN value, .inf: invalid Infinity value"
    },
    "correlation_id": "b16453cf946a4344b2dc9312b697a291",
    "code": 3
}
```

##### Regressions



Based on our metrics, we don't expect that there are any real uses for these values. The support for these values has been deprecated since `v3.25`.

#### Checklist
